docs/operator-manual/user-management/keycloak.md

```diff
@@ -19,62 +28,97 @@ Enable the __Client authentication__.
-using the Default category.
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: argocd-cm
+data:
+  url: https://argocd.example.com
+  oidc.config: |
+    name: Keycloak
+    issuer: https://keycloak.example.com/realms/master
+    clientID: argocd
+    clientSecret: $oidc.keycloak.clientSecret
+    requestedScopes: ["openid", "profile", "email", "groups"]
+```
 
-![Keycloak client scope](../../assets/keycloak-client-scope.png "Keycloak client scope")
+Make sure that:
 
-Create a group called _ArgoCDAdmins_ and have your current user join the group.
+- __issuer__ ends with the correct realm (in this example _master_)
+- __issuer__ on Keycloak releases older than version 17 the URL must include /auth (in this example /auth/realms/master)
+- __clientID__ is set to the Client ID you configured in Keycloak
+- __clientSecret__ points to the right key you created in the _argocd-secret_ Secret
+- __requestedScopes__ contains the _groups_ claim if you didn't add it to the Default scopes
 
-![Keycloak user group](../../assets/keycloak-user-group.png "Keycloak user group")
+## Keycloak and ArgoCD with PKCE
 
-## Configuring ArgoCD OIDC
+These instructions will take you through the entire process of getting your ArgoCD application authenticating with Keycloak.
 
-Let's start by storing the client secret you generated earlier in the argocd secret _argocd-secret_.
+You will create a client within Keycloak and configure ArgoCD to use Keycloak for authentication, using groups set in Keycloak
+to determine privileges in Argo.
 
-1. First you'll need to encode the client secret in base64: `$ echo -n '83083958-8ec6-47b0-a411-a8c55381fbd2' | base64`
-2. Then you can edit the secret and add the base64 value to a new key called _oidc.keycloak.clientSecret_ using `$ kubectl edit secret argocd-secret`.
-   
-Your Secret should look something like this:
+You will also be able to authenticate using argo-cd command line.
 
-```yaml
-apiVersion: v1
-kind: Secret
-metadata:
-  name: argocd-secret
-data:
-  ...
-  oidc.keycloak.clientSecret: ODMwODM5NTgtOGVjNi00N2IwLWE0MTEtYThjNTUzODFmYmQy   
-  ...
-```
+### Creating a new client in Keycloak
+
+First we need to setup a new client.
 
+Start by logging into your keycloak server, select the realm you want to use (`master` by default)
+and then go to __Clients__ and click the __Create client__ button at the top.
+
+![Keycloak add client](../../assets/keycloak-add-client.png "Keycloak add client")
+
+Leave default values.
+
+![Keycloak add client Step 2](../../assets/keycloak-add-client-pkce_2.png "Keycloak add client Step 2")
+
+Configure the client by setting the __Root URL__, __Web origins__, __Admin URL__ to the hostname (https://{hostname}).
+
+Also you can set __Home URL__ to _/applications_ path and __Valid Post logout redirect URIs__ to "https://{hostname}/applications".
+
+The Valid Redirect URIs should be set to:
+- http://localhost:8085/auth/callback (needed for argo-cd cli)
```

Comment:
Is localhost here correct, i.e. not just used for testing?

Reply:
Yes, as when you try to authenticate using sso, argo-cd cli starts a listener on localhost with default port 8085.
It also launcher a web browser where you will fill your login and password.
Then, the callback send your token to argo-cd cli listener.
That's why http://localhost:8085/auth/callback must be allowed.